packages/rocketchat-lib/server/lib/sendNotificationsOnMessage.js

```diff
@@ -162,312 +105,76 @@ RocketChat.callbacks.add('afterSaveMessage', function(message, room, userId) {
-			settings.dontNotifyDesktopUsers.push(subscription.u._id);
-		}
-		if (mobilePushNotifications === 'all' && !disableAllMessageNotifications) {
-			settings.alwaysNotifyMobileUsers.push(subscription.u._id);
-		} else if (mobilePushNotifications === 'nothing') {
-			settings.dontNotifyMobileUsers.push(subscription.u._id);
-		}
-
-		settings.audioNotificationValues[subscription.u._id] = subscription.audioNotificationValue;
-		settings.desktopNotificationDurations[subscription.u._id] = subscription.desktopNotificationDuration;
-
-		if (subscription.muteGroupMentions) {
-			settings.dontNotifyUsersOnGroupMentions.push(subscription.u._id);
-		}
-	});
-	let userIdsForAudio = [];
-	let userIdsToNotify = [];
-	let userIdsToPushNotify = [];
-	const mentions = [];
-	const alwaysNotifyMobileBoolean = RocketChat.settings.get('Notifications_Always_Notify_Mobile');
-
-	const usersWithHighlights = RocketChat.models.Users.findUsersByUsernamesWithHighlights(room.usernames, { fields: { '_id': 1, 'settings.preferences.highlights': 1 }}).fetch()
-		.filter(user => messageContainsHighlight(message, user.settings.preferences.highlights));
-
-	let push_message = ' ';
-	//Set variables depending on Push Notification settings
-	if (RocketChat.settings.get('Push_show_message')) {
-		push_message = parseMessageText(message, userId);
-	}
-
-	let push_username = '';
-	let push_room = '';
-	if (RocketChat.settings.get('Push_show_username_room')) {
-		push_username = user.username;
-		push_room = `#${ room.name }`;
-	}
-
-	if (room.t == null || room.t === 'd') {
-		const userOfMentionId = message.rid.replace(message.u._id, '');
-		const userOfMention = RocketChat.models.Users.findOne({
-			_id: userOfMentionId
-		}, {
-			fields: {
-				username: 1,
-				statusConnection: 1
-			}
-		});
-
-		// Always notify Sandstorm
-		if (userOfMention != null) {
-			RocketChat.Sandstorm.notify(message, [userOfMention._id],
-				`@${ user.username }: ${ message.msg }`, 'privateMessage');
-
-			if (canBeNotified(userOfMentionId, 'desktop')) {
-				const duration = settings.desktopNotificationDurations[userOfMention._id];
-				notifyDesktopUser(userOfMention._id, user, message, room, duration);
-			}
-
-			if (canBeNotified(userOfMentionId, 'mobile')) {
-				if (Push.enabled === true && (userOfMention.statusConnection !== 'online' || alwaysNotifyMobileBoolean === true)) {
-					RocketChat.PushNotification.send({
-						roomId: message.rid,
-						username: push_username,
-						message: push_message,
-						badge: getBadgeCount(userOfMention._id),
-						payload: {
-							host: Meteor.absoluteUrl(),
-							rid: message.rid,
-							sender: message.u,
-							type: room.t,
-							name: room.name
-						},
-						usersTo: {
-							userId: userOfMention._id
-						},
-						category: canSendMessageToRoom(room, userOfMention.username) ? CATEGORY_MESSAGE : CATEGORY_MESSAGE_NOREPLY
-					});
-					return message;
-				}
-			}
-		}
 
+	let subscriptions;
+	if (disableAllMessageNotifications) {
```

Comment:
the find bellow is crucial. all subscription records return will receive at least one kind of notification.

the `query` is defined by the server's default values and `Notifications_Max_Room_Members` setting.

the diff is not helping on this file. but if you look at the entire file, you'll see its logic is very simple and easy to read.